File: DynamicProgramming/53_MaximumSubarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        minv = nums[0]
        ret = minv
        for i in range(1, len(nums)):
            minv = nums[i] if minv <= 0 else minv + nums[i]
            ret = minv if minv > ret else ret
        return ret

File: 53. Maximum Subarray.java
public class Solution {
    public int maxSubArray(int[] nums) {
        if (nums.length == 0) return 0;
        int maxCurr = nums[0], maxSofar = nums[0];
        for (int i = 1; i < nums.length; i++) {
            maxCurr = maxCurr >= 0 ? maxCurr + nums[i] : nums[i];
            maxSofar = Math.max(maxSofar, maxCurr);
        }
        return maxSofar;
    }
}